orchagent/qosorch.cpp

```diff
@@ -284,32 +285,95 @@ bool WredMapHandler::convertFieldValuesToAttributes(KeyOpFieldsValuesTuple &tupl
+           {
+               // set min threshold to the same value as MAX
+               attr.id = SAI_WRED_ATTR_YELLOW_MIN_THRESHOLD;
+               attr.value.s32 = stoi(fvValue(*i));
+               attribs.push_back(attr);
+           }
+           else
+           {
+               min_thresh->value.s32 = stoi(fvValue(*i));
+           }
         }
         else if (fvField(*i) == green_max_threshold_field_name)
         {
             attr.id = SAI_WRED_ATTR_GREEN_MAX_THRESHOLD;
             attr.value.s32 = stoi(fvValue(*i));
             attribs.push_back(attr);
 
-            // set min threshold to the same value as MAX
-            attr.id = SAI_WRED_ATTR_GREEN_MIN_THRESHOLD;
-            attr.value.s32 = stoi(fvValue(*i));
-            attribs.push_back(attr);
+            auto min_thresh = find_if(attribs.begin(), attribs.end(),
+                                [](sai_attribute_t& a) -> bool { return a.id == SAI_WRED_ATTR_GREEN_MIN_THRESHOLD; });
+            if (min_thresh == attribs.end())
+            {
+                // set min threshold to the same value as MAX
+                attr.id = SAI_WRED_ATTR_GREEN_MIN_THRESHOLD;
+                attr.value.s32 = stoi(fvValue(*i));
+                attribs.push_back(attr);
+            }
+        }
+        else if (fvField(*i) == green_min_threshold_field_name)
+        {
+           auto min_thresh = find_if(attribs.begin(), attribs.end(),
+                   [](sai_attribute_t& a) -> bool { return a.id == SAI_WRED_ATTR_GREEN_MIN_THRESHOLD; });
+           if (min_thresh == attribs.end())
+           {
+               // set min threshold to the same value as MAX
+               attr.id = SAI_WRED_ATTR_GREEN_MIN_THRESHOLD;
+               attr.value.s32 = stoi(fvValue(*i));
+               attribs.push_back(attr);
+           }
+           else
+           {
+               min_thresh->value.s32 = stoi(fvValue(*i));
+           }
         }
         else if (fvField(*i) == red_max_threshold_field_name)
         {
             attr.id = SAI_WRED_ATTR_RED_MAX_THRESHOLD;
             attr.value.s32 = stoi(fvValue(*i));
             attribs.push_back(attr);
 
-            // set min threshold to the same value as MAX
-            attr.id = SAI_WRED_ATTR_RED_MIN_THRESHOLD;
-            attr.value.s32 = stoi(fvValue(*i));
-            attribs.push_back(attr);
+            auto min_thresh = find_if(attribs.begin(), attribs.end(),
+                                [](sai_attribute_t& a) -> bool { return a.id == SAI_WRED_ATTR_RED_MIN_THRESHOLD; });
+            if (min_thresh == attribs.end())
+            {
+                // set min threshold to the same value as MAX
+                attr.id = SAI_WRED_ATTR_RED_MIN_THRESHOLD;
+                attr.value.s32 = stoi(fvValue(*i));
+                attribs.push_back(attr);
+            }
+        }
+        else if (fvField(*i) == red_min_threshold_field_name)
+        {
+           auto min_thresh = find_if(attribs.begin(), attribs.end(),
```

Comment:
We don't know in which order we are getting min and max values. If we got max value before min we should change default. Anyway I'm going to add min as required value and will change validation logic